### PR TITLE
feat(controller): support panel api

### PR DESCRIPTION
**Pull request by jialeicui**
## Description

cc @waynelwz 

New APIs are:

| api | method | desc |
|---|---|---|
|/panel/plugin| POST| upload plugin tarball |
| /panel/plugin/{id} | DELETE | remove plugin by id|
| /panel/plugin| GET | list plugins |
| /panel/setting/{project_id}/{key} | GET | get setting by project and key|
| /panel/setting/{project_id}/{key} | GET | save setting by project and key (overwritable)|

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/mapper/PanelSettingMapper.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.apache.ibatis.annotations.Insert;
+import org.apache.ibatis.annotations.Mapper;
+import org.apache.ibatis.annotations.Select;
+
+
+@Mapper
+public interface PanelSettingMapper {
+    @Select("select content from panel_setting where project_id=#{projectId} and name=#{name}")
+    String get(long projectId, String name);
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/panel/mapper/PanelSettingMapper.java:27]: `userId` shoud be in the parameters or this sql will get multiple `contents`  which will cause error

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PanelSettingService.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public void saveSetting(String projectIdStr, String name, String content) {
+        User user = userService.currentUserDetail();
+        var projectId = idConvertor.revert(projectIdStr);
+        panelSettingMapper.set(user.getId(), projectId, name, content);
+    }
+
+    public String getSetting(String projectId, String name) {
+        var id = idConvertor.revert(projectId);
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PanelSettingService.java:52]: current user should be a filter parameter

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            if (content == null) {
+                throw new StarwhaleApiException(new SwProcessException(SwProcessException.ErrorType.SYSTEM),
+                    HttpStatus.BAD_REQUEST);
+            }
+            manifest = yamlMapper.readValue(content, PanelPlugin.PluginManifest.class);
+
+            // check if exists
+            var plugin = panelPluginMapper.getByNameAndVersion(manifest.name, manifest.version);
+            if (plugin != null) {
+                throw new StarwhaleApiException(new SwProcessException(SwProcessException.ErrorType.DB),
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java:89]: this seems to be a validation exception(something wrong with user's input). `SwProcessException` means internal error(something wrong with your code or system) which will cause http code 500.

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        } catch (IOException e) {
+            throw new StarwhaleApiException(new SwProcessException(SwProcessException.ErrorType.STORAGE),
+                HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        // 2.update db
+        var entity = PanelPluginEntity.builder()
+                .name(manifest.name)
+                .version(manifest.version)
+                .meta(new String(content, StandardCharsets.UTF_8))
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java:104]: saving `path`  in `PanelPluginEntity` will bring one benefit: 
- if the code inside `storagePathCoordinator.allocatePluginPath` updates, the `path` could also be found

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/mapper/PanelSettingMapper.java (hunk 1 above)
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/panel/mapper/PanelSettingMapper.java:27]: The PK is `name` and `project_id`, `user_id` is just the origin info.

Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/panel/mapper/PanelSettingMapper.java:27]: We do not support diff eval result views for each user for now.

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java (hunk 4 above)
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java:104]: OK

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java (hunk 3 above)
Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java:89]: We will response 500 even if I set the http status `HttpStatus.CONFLICT`?

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/mapper/PanelSettingMapper.java (hunk 1 above)
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/panel/mapper/PanelSettingMapper.java:27]: OK

File: server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java (hunk 3 above)
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java:89]: > We will response 500 even if I set the http status `HttpStatus.CONFLICT`?

will response `HttpStatus.CONFLICT` here. but SwProcessException is designed to be system error.

Comment (jialeicui) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/panel/PluginService.java:89]: OK, I get.

**Review by anda-ren — APPROVED** [review]
lgtm

**Pull request merged**